sql_alchemy_project8/project8/project8/sqlalchemy_proj8.py

```diff
@@ -0,0 +1,261 @@
+            'AccessionNumber', 'Classification', 'Department',
+            'Cataloged', 'ObjectID']
+artwork_df = pd.read_csv(ARTWORK_CSV, usecols=[col for col in columns2])
+
+
+# connecting to database
+def create_database_engine(postgres_key):
+    '''
+    this function creates a database engine
+    '''
+    try:
+        database_engine = create_engine(postgres_key, echo=True)
+        logging.info("Database engine created successfully.")
+        return database_engine
+    except Exception as e:
+        logging.info(f"Error creating database engine: {e}")
+        return None
+
+
+engine = create_database_engine(POSTGRES_KEY)
+
+Base = declarative_base()
+
+
+# creating table in database
+
+class Artist(Base):
+    __tablename__ = "Artist"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    DisplayName: Mapped[str] = mapped_column(primary_key=True)
+    ArtistBio: Mapped[str]
+    Nationality: Mapped[Optional[str]]
+    Gender: Mapped[Optional[str]]
+    BeginDate = mapped_column(Integer)
+    EndDate = mapped_column(Integer)
+
+    __table_args__ = (
+        Index('my_index', "Nationality", "Gender", postgresql_using='btree'),
+    )
+
+    def __repr__(self):
+        return f"DisplayName({self.DisplayName}), ArtistBio({self.ArtistBio}), Nationality({self.Nationality}), Gender({self.Gender})"
+
+
+class Artwork(Base):
+    __tablename__ = "Artwork"
+
+    Artist = Column(String)
+    ConstituentID = Column(String)
+    ArtistBio = Column(String)
+    Nationality = Column(String)
+    Gender = Column(String)
+    Medium = Column(String)
+    Dimensions = Column(String)
+    CreditLine = Column(String)
+    AccessionNumber = Column(String)
+    Classification = Column(String)
+    Department = Column(String)
+    Cataloged = Column(String)
+    ObjectID = Column(String)
+
+    __table_args__ = (
+        PrimaryKeyConstraint('AccessionNumber', 'ObjectID'),
+        Index("idx_accessionnumber_objectid", 'Department', 'Nationality', postgresql_using='btree')
+    )
+
+
+Base.metadata.create_all(engine)
+
+
+def validate_artist_df(data_):
+    """function checks that each
+    element in each row is of expected
+    datatype """
+
+    # creating copy of data
+    data = data_.copy()
+
+    # loop through data and check if each is of expected data type
+    for num in range(len(data)):
+        if (
+                isinstance(data.loc[num, "DisplayName"], str) and
+                isinstance(data.loc[num, "ArtistBio"], str) and
+                isinstance(data.loc[num, "Nationality"], str) and
+                isinstance(data.loc[num, "Gender"], str) and
+                all(data.loc[num, :].notna())
+        ):
+
+            try:
+                data.loc[num, ['BeginDate', 'EndDate']] = data.loc[num, ['BeginDate', 'EndDate']].astype(int)
+            except:
+                data = data.drop(index=num)
+
+                logging.error(f"Invalid Data Type in row")
+                continue
```

Comment:
if you want to continue, your logger should be `debug`  or `warning` not `error` - error means you wanna stop all you're doing to give attention